rtv 1.11.0 exits on start

Just after start, the start screen is shown for some milliseconds, then the following backtrace appears:

Traceback (most recent call last):
  File "/usr/local/bin/rtv", line 9, in <module>
    load_entry_point('rtv==1.11.0', 'console_scripts', 'rtv')()
  File "/usr/local/lib/python3.4/site-packages/pkg_resources/__init__.py", line 542, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/local/lib/python3.4/site-packages/pkg_resources/__init__.py", line 2569, in load_entry_point
    return ep.load()
  File "/usr/local/lib/python3.4/site-packages/pkg_resources/__init__.py", line 2229, in load
    return self.resolve()
  File "/usr/local/lib/python3.4/site-packages/pkg_resources/__init__.py", line 2235, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/local/lib/python3.4/site-packages/rtv/__main__.py", line 157, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.4/site-packages/rtv/__main__.py", line 138, in main
    page.loop()
  File "/usr/local/lib/python3.4/site-packages/rtv/page.py", line 70, in loop
    self.draw()
  File "/usr/local/lib/python3.4/site-packages/rtv/page.py", line 285, in draw
    self._draw_content()
  File "/usr/local/lib/python3.4/site-packages/rtv/page.py", line 386, in _draw_content
    attr = self._draw_item(subwindow, data, subwin_inverted)
  File "/usr/local/lib/python3.4/site-packages/rtv/subreddit.py", line 222, in _draw_item
    self.term.add_line(win, text, row, 1, curses.A_BOLD)
  File "/usr/local/lib/python3.4/site-packages/rtv/terminal.py", line 273, in add_line
    window.addstr(row, col, text, *params)
_curses.error: addstr() returned ERR

Not sure what other

output from testsuite if it helps:
<pre>
======================================================= FAILURES =======================================================
______________________________________________ test_open_link_subprocess _______________________________________________

terminal = <rtv.terminal.Terminal object at 0x2a1e63ac>

    def test_open_link_subprocess(terminal):
    
        url = 'http://www.test.com'
    
        with mock.patch('time.sleep'),                            \
                mock.patch('os.system'),                          \
                mock.patch('six.moves.input') as six_input,       \
                mock.patch.object(terminal, 'get_mailcap_entry'):
    
            six_input.return_values = 'y'
    
            def reset_mock():
                six_input.reset_mock()
                os.system.reset_mock()
                terminal.stdscr.subwin.addstr.reset_mock()
    
            def get_error():
                # Check if an error message was printed to the terminal
                status = 'Program exited with status'.encode('utf-8')
                return any(status in args[0][2] for args in
                           terminal.stdscr.subwin.addstr.call_args_list)
    
            # Non-blocking success
            reset_mock()
            entry = ('echo ""', 'echo %s')
            terminal.get_mailcap_entry.return_value = entry
            terminal.open_link(url)
            assert not six_input.called
            assert not get_error()
    
            # Non-blocking failure
            reset_mock()
            entry = ('fake .', 'fake %s')
            terminal.get_mailcap_entry.return_value = entry
            terminal.open_link(url)
            assert not six_input.called
>           assert get_error()
E           assert <function test_open_link_subprocess.<locals>.get_error at 0x2a054974>()

tests/test_terminal.py:421: AssertionError
</pre>

--- a/rtv/__main__.py
+++ b/rtv/__main__.py
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
+from __future__ import print_function
 
 import os
 import sys
@@ -39,8 +40,6 @@ def main():
     if six.PY3:
         # These ones get triggered even when capturing warnings is turned on
         warnings.simplefilter('ignore', ResourceWarning)  #pylint:disable=E0602
-
-    locale.setlocale(locale.LC_ALL, '')
 
     # Set the terminal title
     if os.getenv('DISPLAY'):
@@ -102,6 +101,15 @@ def main():
     else:
         # Add an empty handler so the logger doesn't complain
         logging.root.addHandler(logging.NullHandler())
+
+    # Make sure the locale is UTF-8 for unicode support
+    locale.setlocale(locale.LC_ALL, '')
+    encoding = locale.getlocale()[1] or locale.getdefaultlocale()[1]
+    if not encoding or encoding.lower() != 'utf-8':
+        text = ('System encoding was detected as (%s) instead of UTF-8'
+                ', falling back to ascii only mode' % encoding)
+        warnings.warn(text)
+        config['ascii'] = True
 
     # Construct the reddit user agent
     user_agent = docs.AGENT.format(version=__version__)

--- a/rtv/page.py
+++ b/rtv/page.py
@@ -331,7 +331,7 @@ class Page(object):
 
         if os.getenv('DISPLAY'):
             title += ' - rtv {0}'.format(__version__)
-            sys.stdout.write('\x1b]2;{0}\x07'.format(title))
+            sys.stdout.write('\x1b]2;{0}\x07'.format(self.term.clean(title)))
             sys.stdout.flush()
 
         if self.reddit.user is not None:

